mux.go

```diff
@@ -327,6 +336,19 @@ func mapFromPairs(pairs ...string) (map[string]string, error) {
 	return m, nil
 }
 
+func mapFromPairsToRegex(pairs ...string) (map[string]*regexp.Regexp, error) {
+	length, err := checkPairs(pairs...)
+	if err != nil {
+		return nil, err
+	}
+	m := make(map[string]*regexp.Regexp, length/2)
+	for i := 0; i < length; i += 2 {
+		regex, _ := regexp.Compile(pairs[i+1])
```

Comment:
I guess you can just return the error, that will be more in line with the rest of the package's error handling (which badly needs an overhaul, ugh)